app/src/components/fields/CustomNumberField.tsx

```diff
@@ -0,0 +1,77 @@
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { useFormikContext } from 'formik';
+import get from 'lodash-es/get';
+import React from 'react';
+import NumberFormat, { NumberFormatProps } from 'react-number-format';
+
+export type INumberFieldProps = TextFieldProps & {
+  required?: boolean;
+  label: string;
+  name: string;
+  min?: number;
+  max?: number;
+  float?: boolean;
+};
+
+interface NumberFormatCustomProps {
+  onChange: (event: { target: { name: string; value: number } }) => void;
+  name: string;
+  min?: number;
+  max?: number;
+  float?: boolean;
+}
+
+const NumberFormatCustom = React.forwardRef<NumberFormatProps, NumberFormatCustomProps>(function NumericFormatCustom(
+  props,
+  ref
+) {
+  const { onChange, min, max, float = false, ...other } = props;
```

Comment:
This always overrides the float to false. It's probably better to just spread as is, undefined will be falsy anyways.